src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/leaked-passwords/LeakedPasswordsLayout.tsx

```diff
@@ -89,21 +88,20 @@ export function LeakedPasswordsLayout(props: LeakedPasswordsLayoutProps) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [unresolvedPasswordBreach]);
 
-  const resolutionArray: Array<HibpBreachDataTypes[keyof HibpBreachDataTypes]> =
-    unresolvedPasswordBreach.resolvedDataClasses;
-
   const handleUpdateBreachStatus = async () => {
     if (!emailAffected) {
       return;
     }
 
     try {
-      resolutionArray.push(resolvedDataClassName);
-
+      unresolvedPasswordBreach.resolvedDataClasses.push(resolvedDataClassName);
+      // Remove empty [""] string
+      const formattedDataClasses =
+        unresolvedPasswordBreach.resolvedDataClasses.filter(Boolean);
```

Comment:
Thanks, this would filter out the "", though I'd like to understand where the empty string is coming from and perhaps fix it there (in case other parts of our code also use it and run into the same issue)